### PR TITLE
Implement MediaRemoteOptionDialog and apply it for YouTube caption

### PR DESCRIPTION
**Pull request by romandev**
Implement MediaRemoteOptionDialog for YouTube features such as caption.
The roles of each class are as follows:
  - **MediaRemoteOptionDialog**: This is a common container of single
    choice items dialog. We can fill the list using DataModel.
  - **DataModel**: The data model represents the list to be filled in
    the UI.
  - **YouTubeOptionDialogDelegate**: Query data via YouTube APIs and
    create data models and dialogs.

Refs: #884

### CODE REVIEW

File: src/org/triple/banana/media_remote/YouTubeOptionDialogDelegate.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import org.triple.banana.youtube.YouTubePlayerController;
+
+class YouTubeOptionDialogDelegate {
+    private final @NonNull MediaRemoteOptionDialog mDialog;
```
Comment (jdragonbae) [inline src/org/triple/banana/media_remote/YouTubeOptionDialogDelegate.java:16]: Does this dialog dismiss when entering pip mode?

File: src/org/triple/banana/media_remote/MediaRemoteService.java [hunk 2]
```diff
@@ -6,12 +6,16 @@
 package org.triple.banana.media_remote;
 
 import android.app.Activity;
+import android.content.Context;
 import android.content.pm.ActivityInfo;
 import android.content.res.Configuration;
 import android.os.Build;
 import android.os.Handler;
+import android.util.Log;
```
Comment (jdragonbae) [inline src/org/triple/banana/media_remote/MediaRemoteService.java:14]: Looks unused import.

File: src/org/triple/banana/media_remote/YouTubeOptionDialogDelegate.java (hunk 1 above)
Comment (romandev) [reply src/org/triple/banana/media_remote/YouTubeOptionDialogDelegate.java:16]: Done.

File: src/org/triple/banana/media_remote/MediaRemoteService.java (hunk 2 above)
Comment (romandev) [reply src/org/triple/banana/media_remote/MediaRemoteService.java]: Done.

**Review by jdragonbae — APPROVED** [review]
lgtm

**Pull request merged**